tools/integration_tests/util/operations/file_operations.go

```diff
@@ -273,3 +277,160 @@ func ReadChunkFromFile(filePath string, chunkSize int64, offset int64) (chunk []
 
 	return
 }
+
+// Returns the stats of a file.
+// Fails if the passed input is a directory.
+func StatFile(file string) (*fs.FileInfo, error) {
+	fstat, err := os.Stat(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to stat input file %s: %v", file, err)
+	} else if fstat.IsDir() {
+		return nil, fmt.Errorf("input file %s is a directory", file)
+	}
+
+	return &fstat, nil
+}
+
+// Finds if two local files have identical content (equivalnt to binary diff).
+// Needs (a) both files to exist, (b)read permission on both the files, (c) both
+// inputs to be proper files, symlinks/directories not supported.
+// Compares file names first. If different, compares sizes next.
+// If sizes match, then compares hashes of both the files.
+// Not a good idea for very large files as it loads both the files in the memory completely.
+// Returns 0 if no error and files match.
+// Returns 1 if files don't match and captures reason for mismatch in err.
+// Returns 2 if any error.
+func DiffFiles(filepath1, filepath2 string) (int, error) {
```

Comment:
I believe this function is not used in the current tests ? If yes, it might be good to add it in a separate PR ?

Reply:
My intention was to use it in the full-file read tests to diff the content in mounted directory with the content in the gcs bucket. I'll add those calls.

Reply:
Done. Being used in the gzip tests now.